src/hb-ot-hmtx-table.hh

```diff
@@ -190,33 +190,19 @@ struct hmtxvmtx
 		      unsigned int default_advance_ = 0)
     {
       default_advance = default_advance_ ? default_advance_ : hb_face_get_upem (face);
+      ascender = 0; descender = 0; line_gap = 0;
 
-      bool got_font_extents = false;
-      if (T::os2Tag != HB_TAG_NONE)
-      {
-#define USE_TYPO_METRICS (1u<<7)
-	if (0 != (face->table.OS2->fsSelection & USE_TYPO_METRICS))
-	{
-	  ascender = abs (face->table.OS2->sTypoAscender);
-	  descender = -abs (face->table.OS2->sTypoDescender);
-	  line_gap = face->table.OS2->sTypoLineGap;
-	  got_font_extents = (ascender | descender) != 0;
-	}
-      }
+      /* XXX Shouldn't create a font */
+      hb_font_t *font = hb_font_create (face);
+      hb_ot_metrics_get_position (font, (hb_ot_metrics_t) T::ascenderTag, &ascender);
+      hb_ot_metrics_get_position (font, (hb_ot_metrics_t) T::descenderTag, &descender);
+      hb_ot_metrics_get_position (font, (hb_ot_metrics_t) T::lineGapTag, &line_gap);
+      ascender = abs (ascender);
+      descender = -abs (descender);
```

Comment:
Umm.  I wonder if this abs business should happen in the metrics_get_position()...  We want that API and the `font_extents` API to produce same values, no?